servicetalk-http-api/src/main/java/io/servicetalk/http/api/DefaultHttpRequestMetaData.java

```diff
@@ -356,6 +366,79 @@ private Uri lazyParseRequestTarget() {
         return requestTargetUri;
     }
 
+    private void query(final Map<String, List<String>> params) {
+        Uri httpUri = lazyParseRequestTarget();
+        StringBuilder sb = new StringBuilder(httpUri.uri().length() + params.size() * 8);
+
+        appendScheme(sb, httpUri);
+        appendAuthority(sb, httpUri);
+        sb.append(httpUri.path());
+
+        // Append query params
+        Iterator<Entry<String, List<String>>> itr = params.entrySet().iterator();
+        char prefixChar = '?';
+        while (itr.hasNext()) {
+            Entry<String, List<String>> next = itr.next();
+            String encodedKey = encodeComponent(QUERY, next.getKey(), REQUEST_TARGET_CHARSET, true);
+            sb.append(prefixChar).append(encodedKey);
+            List<String> values = next.getValue();
+            if (values != null) {
```

Comment:
the `null` check and method name `setQueryParams` -> `query` are the only changes in this method.